scripts/test.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+# This script is executed by Travis CI for every successful push (on any branch, PR or not).
+# It runs some basic tests on pages. If the build is also a PR, additional
+# checks are run through the check-pr script, and any message or error is sent
+# to tldr-bot to be commented on the PR.
+#
+# NOTE: must be run from the repository root directory to correctly work!
+# NOTE: `set -e` is applied conditionally only if needed.
+
+# Default test.
+function default_test {
+  markdownlint pages*/**/*.md
+  tldr-lint ./pages
+}
+
+# Special test for Travis CI pull request builds.
+# Runs default_test collecting errors for tldr-bot.
+function pr_test {
+  errs=`default_test 2>&1`
+
+  if [ -n "$errs" ]; then
+    echo -e "Test failed!\n$errs\n" >&2
+    echo 'Sending errors to tldr-bot.' >&2
+    echo -n "$errs" | python3 scripts/send_to_bot.py error
+    exit 1
+  fi
+}
+
+# Additional checks for Travis CI pull request builds.
+# Only taken as suggestions, does not make the build fail.
+function pr_check {
+  msgs=`bash scripts/check-pr.sh`
+
+  if [ -n "$msgs" ]; then
+    echo -e "\nCheck PR reported the following message(s):\n$msgs\n" >&2
+    echo 'Sending check results to tldr-bot.' >&2
+    echo -n "$msgs" | python3 scripts/send_to_bot.py check
+  fi
+}
+
+###################################
+# MAIN
+###################################
+
+if [ "$TRAVIS" = "true" ] && [ "$TRAVIS_REPO_SLUG" = "tldr-pages/tldr" ] && [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+  pr_check
```

Comment:
I think if you refactor the 2 lines I suggested above, you don't need separate functions for these. Just have functions for default_test (probably can rename it to test now), and the send_to_bot.

Reply:
I think this would hurt readability. I split the code in two functions to clearly define what is a test (that can fail the build) and what is just a check (that can generate warnings but not fail the build).

Reply:
The default_test function will still be there. You will just be unwrapping pr_test and pr_check. When I read the code, it wasn't immediately clear to me that `pr_test` wraps `default_test`. Hence I suggested this. If you still want to do it, I would suggest to use names like `test` and `wrapped_test` or something similar.

Reply:
I avoided naming the function exactly `test` because that would have shadowed the `test` command and it seemed like a bad idea :confused: 

`default_test` is the first name that popped into my mind, probably not *the best* name though, but right now I can't think of something else that's better other than just `test`... I've got to think harder :sweat_smile:

Reply:
I think I'd prefer them separate personally.

As for a name, what about `default_test` -> `run_tests` and `pr_test` -> `run_tests_travis`?

I'd argue that the 2nd function doesn't have any additional checks at all - it's just a wrapper for handling the error messages differently.

Reply:
I like the names, and yes that's exactly the purpose of the second function.

Reply:
@agnivade what do you think about naming the functions `run_tests`, `run_tests_travis_pr` and `run_checks_travis_pr`?

Reply:
Sounds good to me. Is there a reason to add "_travis_" ? Seems to me that "run_tests_pr" is sufficient.

Reply:
No particular reason, except to make it more explanatory.